spec/rspec/its_spec.rb

```diff
@@ -376,5 +378,58 @@ def terminator
         end
       end
     end
+
+    describe "#fits" do
+      it 'calls #its with :focus metadata' do
+        stub = Class.new do
+          include RSpec::Its
+        end.new
+        allow(stub).to receive(:its)
+
+        expect(stub).to receive(:its).with('attribute', :focus).and_yield
+        stub.fits('attribute') {}
+      end
+
+      it "fits is focused within the example group" do
+        ex_its, ex_fits = nil
+
+        RSpec.describe do
+          extend RSpec::Its
+
+          ex_its = its(:name) { :its_example }
+          ex_fits = fits(:class) { :fits_example }
+        end
+
+        expect(ex_its.examples.first.metadata[:focus]).to be_falsey
+        expect(ex_fits.examples.first.metadata[:focus]).to be_truthy
+      end
+    end
+
+    describe "#xits" do
+      it 'calls #its with :skip metadata' do
+        stub = Class.new do
+          include RSpec::Its
+        end.new
+        allow(stub).to receive(:its)
+
+        expect(stub).to receive(:its).with('attribute', :skip).and_yield
+        stub.xits('attribute') {}
+      end
+
+      it "xits is skipped within the example group" do
+        ex_its, ex_xits = nil
+
+        RSpec.describe do
+          extend RSpec::Its
+
+          ex_its = its(:name) { :its_example }
+          ex_xits = xits(:class) { :xits_example }
+        end
+
+        expect(ex_its.examples.first.metadata[:skip]).to be_falsey
+        expect(ex_xits.examples.first.metadata[:skip]).to be_truthy
```

Comment:
FYI `#its` returns a RSpec `describe` block rather than the actual example, which is why we need to select the first example if that describe block (our its example)
https://github.com/rspec/rspec-its/pull/79/files#diff-39939cafdb9cee9c4840e8adc3819e0a45dafa210f75a51a34ed5bdb74c7ef4fL123

Reply:
Isn't metadata applied to this example group itself? Can you check example group metadata instead?

Reply:
That doesn't appear to be the case:
```ruby
pry(#<RSpec::ExampleGroups::RSpecIts::Fits>)> ex_fits.metadata
=> {:execution_result=>#<RSpec::Core::Example::ExecutionResult:0x007fbdf61f6268>,
 :block=>#<Proc:0x007fbdf61f64c0@/Users/baron/Developer/personal/rspec-its/lib/rspec/its.rb:123>,
 :description_args=>["class"],
 :description=>"class",
 :full_description=>" class",
 :described_class=>nil,
 :file_path=>"./spec/rspec/its_spec.rb",
 :line_number=>400,
 :location=>"./spec/rspec/its_spec.rb:400",
 :parent_example_group=>
  {:execution_result=>#<RSpec::Core::Example::ExecutionResult:0x007fbdf61fe648>,
   :block=>#<Proc:0x007fbdf61fe8c8@/Users/baron/Developer/personal/rspec-its/spec/rspec/its_spec.rb:396>,
   :description_args=>[nil],
   :description=>"",
   :full_description=>"",
   :described_class=>nil,
   :file_path=>"./spec/rspec/its_spec.rb",
   :line_number=>396,
   :location=>"./spec/rspec/its_spec.rb:396"}}
```
versus the current code:
```ruby
pry(#<RSpec::ExampleGroups::RSpecIts::Fits>)> ex_fits.examples.first.metadata
=> {:execution_result=>#<RSpec::Core::Example::ExecutionResult:0x007fbdf61f4cb0>,
 :block=>#<Proc:0x007fbdf61f6e48@/Users/baron/Developer/personal/rspec-its/spec/rspec/its_spec.rb:400>,
 :description_args=>[],
 :description=>"",
 :full_description=>" class ",
 :described_class=>nil,
 :file_path=>"./spec/rspec/its_spec.rb",
 :line_number=>400,
 :location=>"./spec/rspec/its_spec.rb:400",
 :example_group=>
  {:execution_result=>#<RSpec::Core::Example::ExecutionResult:0x007fbdf61f6268>,
   :block=>#<Proc:0x007fbdf61f64c0@/Users/baron/Developer/personal/rspec-its/lib/rspec/its.rb:123>,
   :description_args=>["class"],
   :description=>"class",
   :full_description=>" class",
   :described_class=>nil,
   :file_path=>"./spec/rspec/its_spec.rb",
   :line_number=>400,
   :location=>"./spec/rspec/its_spec.rb:400",
   :parent_example_group=>
    {:execution_result=>#<RSpec::Core::Example::ExecutionResult:0x007fbdf61fe648>,
     :block=>#<Proc:0x007fbdf61fe8c8@/Users/baron/Developer/personal/rspec-its/spec/rspec/its_spec.rb:396>,
     :description_args=>[nil],
     :description=>"",
     :full_description=>"",
     :described_class=>nil,
     :file_path=>"./spec/rspec/its_spec.rb",
     :line_number=>396,
     :location=>"./spec/rspec/its_spec.rb:396"}},
 :focus=>true}
```

Reply:
You are right, metadata is only set on the example itself https://github.com/rspec/rspec-its/blob/97d9e040f1d98ae6151fafd5895efe2b4d1e8276/lib/rspec/its.rb#L169